pulsar-common/src/main/proto/PulsarApi.proto

```diff
@@ -197,6 +197,10 @@ message CommandConnect {
 message CommandConnected {
 	required string server_version = 1;
 	optional int32 protocol_version = 2 [default = 0];
+
```

Comment:
Break this out into two new messages, AuthChallenge and AuthResponse. As it is now the auth flow is mixed into the connect flow, is it makes things harder to follow.

Reply:
Thanks @ivankelly.  If you are agree, would like to re-use connect command. 
Actually,  auth flow is indeed part of connect flow, and the original auth method is part of the connect logic. 
Reuse the connect command here is also earlier for the backward compatible.  If provide a new AuthData, the `CommandConnect` in above is better to keep align, and change `auth_method_name` and `auth_data` into the new AuthData.

Reply:
@merlimat what do you think of it here?

Reply:
The backward compat story here is that a old client will connect, get a CommandConnected back with the challenge embedded, think it has connected and then hang forever.

Reply:
Thanks @ivankelly. as discussed will add a new pair of command to do mutual auth.